api/src/main/java/com/epam/pipeline/manager/cluster/performancemonitoring/ResourceMonitoringManager.java

```diff
@@ -156,7 +160,26 @@ private void processOverloadedRuns(final List<PipelineRun> runs) {
                 .filter(pod -> isPodUnderPressure(pod.getValue(), thresholds))
                 .collect(Collectors.toList());
 
+        final List<PipelineRun> runsToUpdateTags = getRunsToUpdatePressuredTags(running, runsToNotify);
         notificationManager.notifyHighResourceConsumingRuns(runsToNotify, NotificationType.HIGH_CONSUMED_RESOURCES);
+        pipelineRunManager.updateRunsTags(runsToUpdateTags);
+    }
+
+    private List<PipelineRun> getRunsToUpdatePressuredTags(final Map<String, PipelineRun> running,
+                              final List<Pair<PipelineRun, Map<ELKUsageMetric, Double>>> runsToNotify) {
+        final Set<Long> listOfRunsIdToNotify = runsToNotify
+                .parallelStream()
+                .map(p -> p.getLeft().getId())
+                .collect(Collectors.toSet());
+        final Stream<PipelineRun> runsStream = running.values().parallelStream();
+        final Predicate<PipelineRun> isToBeNotified = r -> listOfRunsIdToNotify.contains(r.getId());
+        final Stream<PipelineRun> runsToAddTag = runsStream
+                .filter(isToBeNotified)
+                .map(r -> r.inverseTagForRun(UTILIZATION_LEVEL_HIGH, TRUE_VALUE_STRING));
```

Comment:
there is something wrong with logic.
we should check that if pipeline would be notified and doesn't have tag we should add it
and if it will be not notified and tag is present - we should delete it